### PR TITLE
QUIC: Wire SSL_net_(read|write)_desired for TLS/DTLS

### PR DESCRIPTION
**Pull request by hlandau**

### CODE REVIEW

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Should this be reflected in manual pages?

**Review by paulidale — APPROVED** [review]

**Comment by mattcaswell** [comment]
Needs a documentation update. The man page for these functions says "These functions are not supported on non-QUIC SSL objects."

**Comment by mattcaswell** [comment]
Should there be a test for this?

**Comment by mattcaswell** [comment]
Ping?? This seems fairly urgent?

**Comment by hlandau** [comment]
Updated

- Implement SSL_get_[rw]poll_descriptor for SSL/DTLS as simple BIO passthrough
- Add tests
- Update documentation

**Review by t8m — DISMISSED** [review]

**Comment by t8m** [comment]
@hlandau CI is relevant

**Comment by hlandau** [comment]
CI passing

**Review by t8m — APPROVED** [review]

File: doc/man3/SSL_get_rpoll_descriptor.pod [hunk 1]
```diff
@@ -66,7 +66,14 @@ may change in response to any call to the SSL object other than
 SSL_net_read_desired(), SSL_net_write_desired(), SSL_get_rpoll_descriptor(),
 SSL_get_wpoll_descriptor() and SSL_get_event_timeout().
 
-These functions are not supported on non-QUIC SSL objects.
+On non-QUIC SSL objects, calls to SSL_get_rpoll_descriptor() and
+SSL_get_wpoll_descriptor() function the same as calls to
+BIO_get_rpoll_descriptor() and BIO_get_wpoll_descriptor() respectively on the
+respective read and write BIOs configured on the SSL object.
```
Comment (mattcaswell) [inline doc/man3/SSL_get_rpoll_descriptor.pod:72]: "respectively on the respective" sounds a bit odd

**Review by mattcaswell — CHANGES_REQUESTED** [review]

**Comment by hlandau** [comment]
Updated.

**Review by mattcaswell — APPROVED** [review]

**Comment by mattcaswell** [comment]
@t8m please reconfirm

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks.

**Pull request closed without merging**